ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallOptions.kt

```diff
@@ -79,8 +79,8 @@ data class PaywallOptions internal constructor(
             this.listener = listener
         }
 
-        fun setMyAppPurchaseLogic(myAppPurchaseLogic: MyAppPurchaseLogic?) = apply {
-            this.myAppPurchaseLogic = myAppPurchaseLogic
+        fun setPurchaseLogic(purchaseLogic: PurchaseLogic?) = apply {
```

Comment:
I honestly don't fully love this naming... Feels a bit too generic for what it does from an API standpoint... I kinda prefer what we had before `setMyAppPurchaseLogic` or maybe `setCustomPurchaseLogic`. Wdyt?

Reply:
I agree that the function name could be too generic. `setCustomPurchaseLogic` could work 🤔

Reply:
I think `setCustomPurchaseLogic` could infer it's always going to set it to `MyApp`, as in set it to use my custom logic. I think I prefer `setMyAppPurchaseLogic`

Reply:
I agree as well.  Removing the `custom` makes it look like it is mandatory to set I think. Also, why is the `purchaseLogic` parameter optional? Why would someone set it to null?

Reply:
Thanks! I've changed the setter to `setCustomPurchaseLogic`.

As for the nullable parameter, I was following the pattern for assigning objects established above it:

```kotlin
        fun setOffering(offering: Offering?) = apply {
            this.offering = offering
        }

        fun setFontProvider(fontProvider: FontProvider?) = apply {
            this.fontProvider = fontProvider
        }

        fun setListener(listener: PaywallListener?) = apply {
            this.listener = listener
        }
```

If they aren't nullable, calling code that currently looks like this:

```kotlin
        PaywallDialogOptions.Builder()
            .setDismissRequest(onDismiss)
            .setOffering(currentState.offering)
            .setFontProvider(currentState.fontProvider)
            .setCustomPurchaseLogic(currentState.purchaseLogic)
            .build()
```

gets a little messier:

```kotlin
        PaywallDialogOptions.Builder().apply {
            setDismissRequest(onDismiss)
            currentState.offering?.let { setOffering(it) }
            currentState.fontProvider?.let { setFontProvider(it) }
            currentState.purchaseLogic?.let { setCustomPurchaseLogic(it) }
        }.build()
```

Reply:
Nullable looks fine to me since it matches the existing methods for consistency 🤷‍♂️